boards/arm/b_l4s5i_iot01a/b_l4s5i_iot01a.yaml

```diff
@@ -0,0 +1,13 @@
+identifier: b_l4s5i_iot01a
+name: ST STM32L4S5I IOT01A Discovery
```

Comment:
ST B_L4S5I_IOT01A Discovery kit
(For doc generation purpose, it should be identical to .rst header)

Reply:
Name is changed to ST B_L4S5I_IOT01A Discovery kit for doc/index.rst and b_l4s5i_iot01a.yaml